### PR TITLE
测试用例已跑过

### PR DESCRIPTION
**Pull request by jiangzhuo**
test passed and I sweared https://github.com/strongloop/express/blob/master/lib/router/route.js#L98 not been referenced.

### CODE REVIEW

**Comment by yorkie** [comment]
Good, and could you please make indention to 2 to keep consistent with other source files? Thanks

File: middleware.js [hunk 1]
```diff
@@ -1,4 +1,45 @@
...
+    this.handlers.push({route: route, handleFn: fn});
 };
+
+app.handle = function (req, res) {
+    var index = 0;
+    var path = url.parse(req.url).pathname;
+    if (undefined == path) path = '/';
+
+    function next() {
+        var handler = app.handlers[index++];
```
Comment (yorkie) [inline middleware.js]: Move the `++` operator to the end

File: middleware.js [hunk 2]
```diff
@@ -1,4 +1,45 @@
...
+
+        if (!handler) {
+            return;
```
Comment (yorkie) [inline middleware.js]: Consider the following call:

```
app.use('/', false);
app.use('/', function (req, res, next) { next() });
```

Then 2nd handler could not be called.

File: middleware.js [hunk 3]
```diff
@@ -1,4 +1,45 @@
...
+        }
+
+        if (handler.route == '/' || handler.route == path.toLowerCase()) {
```
Comment (yorkie) [inline middleware.js]: use `===` please :-)

**Comment by yorkie** [comment]
@jiangzhuo Round 1 is done, basically LGTM except for a little bit nits :-)

File: middleware.js (hunk 2 above)
Comment (jiangzhuo) [inline middleware.js]: Maybe we should update test case firstly.

Comment (yorkie) [inline middleware.js]: Good idea :-)

**Comment by yorkie** [comment]
@jiangzhuo how is it going? we are still waiting for your upcoming fixes, thank you :-)

File: middleware.js (hunk 1 above)
Comment (jiangzhuo) [inline middleware.js]: still dont understand why move ++ to the end

Comment (yorkie) [inline middleware.js]: Notes the following 2 points:
- the `++` is not clear to represent what the statement wants
- we should make sure `index` gets changed iff(if and only if) one `next` function is going to quit.

File: middleware.js [hunk 4]
```diff
@@ -1,4 +1,47 @@
...
-exports.createServer = function () {
+var app = {};
+
+app.use = function (route, fn) {
+  if ('function' == typeof route) {
+    fn = route;
+    route = '/';
+  }
+  // maybe ignore wrong arguments handler
+  if ('function' == typeof fn) {
```
Comment (yorkie) [inline middleware.js]: use strict equation...

File: middleware.js [hunk 5]
```diff
@@ -1,4 +1,47 @@
...
+    this.handlers.push({route: route, handleFn: fn});
+  }
+};
+
+app.handle = function (req, res) {
+  var index = 0;
+  var path = url.parse(req.url).pathname;
+  if (undefined === path) path = '/';
+
+  function next() {
```
Comment (yorkie) [inline middleware.js]: use `function next () {` because you used this style above :-)

File: test.js [hunk 6]
```diff
@@ -103,6 +103,29 @@ describe('middleware', function () {
...
+
+      app.use(function (req, res) {
+        res.end('end');
+      });
+
+      app.listen(3005, function(){
+        var self = this;
+
+        request(self)
+            .get('/')
```
Comment (yorkie) [inline test.js]: indention :-)

**Comment by jiangzhuo** [comment]
I just try my Webstorm formate code function.
I think 'function' == typeof XXXX  use strict equation is  a little overkill

**Comment by yorkie** [comment]
> I think 'function' == typeof XXXX use strict equation is a little overkill

Why do you think so?

**Comment by yorkie** [comment]
> I just try my Webstorm formate code function.

We are not restrict what you should work on, but you should make your code be consistent with our coding style, thank you :-)

**Comment by jiangzhuo** [comment]
typeof always return a string, == can compare two string. But from code style side I agree to use ===, and I just find https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof examples all use ===

**Comment by jiangzhuo** [comment]
I agree to be consistent with previous code's style. I will pay attention to this.

**Comment by yorkie** [comment]
@jiangzhuo thanks so much for patches, this looks good enough, now I will close this pr because it is just for testing :-)

I will connect you next Monday, have a nice weekend :-)

**Comment by jiangzhuo** [comment]
thank you and good night

**Pull request closed without merging**